common/ledger/blockledger/util.go

```diff
@@ -74,6 +78,11 @@ func CreateNextBlock(rl Reader, messages []*cb.Envelope) *cb.Block {
 	block := protoutil.NewBlock(nextBlockNumber, previousBlockHash)
 	block.Header.DataHash = protoutil.BlockDataHash(data)
 	block.Data = data
+	block.Metadata.Metadata[cb.BlockMetadataIndex_SIGNATURES] = protoutil.MarshalOrPanic(&cb.Metadata{
```

Comment:
It's due to the LastConfig now being nested inside the OrdererBlockMetadata instead of as its own separate value. Without setting this here, certain tests panic with a nil pointer when attempting to retrieve the index of a nil LastConfig (in regstrar_test.go's testLastConfigBlockNumber function). It's possibly the case that these tests weren't really testing that the last config was set correctly and merely relied on the default value.

Reply:
Then I think we need to fix those tests.  The `CreateNextBlock` function has no business setting this metadata.  In fact, it can't set it correctly, because it doesn't know whether this block is a config block (in which case, the last config should be this block number.)  As implemented above, all blocks would point to block 0 as the last config, as there is no case where the value is updated.